src/components/common/dialog.tsx

```diff
@@ -0,0 +1,918 @@
+
+   - <DialogTextInput label="Name" value={dlg.value("name")} ... />
+
+   This will produce a TextInput in a (optional) FormGroup that will
+   manage the given value handle.  The "label" property is optional
+   and omitting it will also omit the FormGroup.
+
+  - <DialogCheckbox label= value= .../>
+
+  For a single checkbox that drives a boolean.
+
+  - <DialogRadioSelect label= value= options= .../>
+
+  For a group of radio buttons.  The options can be disabled and have
+  explanations.
+
+ */
+
+import React from "react";
+import { useObject, useOn } from 'hooks';
+import { EventEmitter } from 'cockpit/event';
+
+import cockpit from "cockpit";
+
+import { Button, type ButtonProps } from "@patternfly/react-core/dist/esm/components/Button/index.js";
+import { FormGroup, type FormGroupProps, FormHelperText } from "@patternfly/react-core/dist/esm/components/Form";
+import { TextInput, type TextInputProps } from "@patternfly/react-core/dist/esm/components/TextInput";
+import { Alert } from "@patternfly/react-core/dist/esm/components/Alert/index.js";
+import { HelperText, HelperTextItem } from "@patternfly/react-core/dist/esm/components/HelperText";
+import { Checkbox } from "@patternfly/react-core/dist/esm/components/Checkbox";
+import { FormSelect, type FormSelectProps } from "@patternfly/react-core/dist/esm/components/FormSelect";
+import { Radio } from "@patternfly/react-core/dist/esm/components/Radio";
+
+const _ = cockpit.gettext;
+
+type ArrayElement<ArrayType> =
+  ArrayType extends readonly (infer ElementType)[] ? ElementType : never;
+
+interface DialogValidationState {
+    cached_value: unknown;
+    cached_result: string | undefined;
+    // TODO - all the stuff for async
+}
+
+function toSpliced<T>(arr: T[], start: number, deleteCount: number, ...rest: T[]): T[] {
+    const copy = [...arr];
+    copy.splice(start, deleteCount, ...rest);
+    return copy;
+}
+
+export class DialogValue<T> {
+    /* eslint-disable no-use-before-define */
+    dialog: DialogState<unknown>;
+    /* eslint-enable */
+    getter: () => T;
+    setter: (val: T) => void;
+    path: string;
+
+    constructor(dialog: DialogState<unknown>, getter: () => T, setter: (val: T) => void, path: string) {
+        this.dialog = dialog;
+        this.getter = getter;
+        this.setter = setter;
+        this.path = path;
+    }
+
+    validation_text(): string | undefined {
+        return this.dialog.validation[this.path];
+    }
+
+    get(): T {
+        return this.getter();
+    }
+
+    set(val: T): void {
+        this.setter(val);
+    }
+
+    id(tag: string = "field"): string {
+        return "dialog-" + tag + "-" + this.path;
+    }
+
+    map<X>(func: (val: DialogValue<ArrayElement<T>>, index: number) => X): X[] {
+        const val = this.get();
+        if (Array.isArray(val)) {
+            return val.map((_, i) => func(this.sub(i as keyof T) as DialogValue<ArrayElement<T>>, i));
+        } else
+            return [];
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2435-ec962d21-20251204-104905-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)